src/Hosting/TestHost/src/HttpContextBuilder.cs

```diff
@@ -160,7 +160,17 @@ internal void ClientInitiatedAbort()
 
         private async Task CompleteRequestAsync()
         {
-            if (!_requestPipe.Reader.TryRead(out var result) || !result.IsCompleted)
+            bool requestBodyInProgress;
+            try
+            {
+                requestBodyInProgress = !_requestPipe.Reader.TryRead(out var result) || !result.IsCompleted;
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException("An error occurred when completing the request. Request delegate may have finished while there is a pending read of the request body.", ex);
```

Comment:
What's the inner exception type and message in this scenario?

Reply:
> System.InvalidOperationException: Reading is already in progress.

I didn't want to completely throw away the original exception in case there is some scenario where an exception is thrown for a different reason.

Reply:
I'm not sure this is a substantial improvement. The exception type doesn't change, and the new message can't say with any confidence how to fix the problem.

Was there an issue associated with this PR? Was the IOE even reported to the client/test or was it swallowed internally?

Reply:
Issue: https://github.com/grpc/grpc-dotnet/issues/658

The current error isn't descriptive of the problem and how the developer could fix it.